tests/saitests/sai_qos_tests.py

```diff
@@ -579,6 +580,112 @@ def runTest(self):
+        src_port_slice = self.test_params['src_port_slice']
+        dst_port_id = self.test_params['dst_port_id']
+        dst_port_ip = self.test_params['dst_port_ip']
+
+        dscp_to_tc_map = self.test_params['tunnel_qos_map']['dscp_to_tc_map']
+        tc_to_pg_map = self.test_params['tunnel_qos_map']['tc_to_priority_group_map']
+        asic_type = self.test_params['sonic_asic_type']
+        dscp_to_pg_map = {}
+        PKT_NUM = 100
+        # We noticed that a packet less than 256 bytes will occupy a 256-sized buffer
+        FACTOR = 256
+        # There is background traffic during test, so we need to add error tolerance to ignore such pakcets
+        ERROR_TOLERANCE = {
+            0: 10,
+            1: 0,
+            2: 0,
+            3: 0,
+            4: 0,
+            5: 0,
+            6: 0,
+            7: 0
+        }
+        for dscp, tc in dscp_to_tc_map.items():
+            dscp_to_pg_map[dscp] = tc_to_pg_map[tc]
+        try:
+            # Disable tx on EGRESS port so that headroom buffer cannot be free
+            sai_thrift_port_tx_disable(self.client, asic_type, [dst_port_id])
+
+            # There are packet leak even port tx is disabled (18 packets leak on TD3 found)
+            # Hence we send some packet to fill the leak before testing
+            for dscp, _ in dscp_to_pg_map.items():
+                pkt = self._build_testing_pkt(
+                                            active_tor_mac=active_tor_mac,
+                                            standby_tor_mac=standby_tor_mac,
+                                            active_tor_ip=active_tor_ip,
+                                            standby_tor_ip=standby_tor_ip,
+                                            inner_dscp=dscp,
+                                            outer_dscp=0,
+                                            dst_ip=dst_port_ip
+                                        )
+                send_packet(self, src_port_id, pkt, 20)
+            time.sleep(10)
+
+            for dscp, pg in dscp_to_pg_map.items():
+                # Build and send packet to active tor.
+                # The inner DSCP is set to testing value, and the outer DSCP is set to 0 as it has no impact on remapping
+                pkt = self._build_testing_pkt(
+                                            active_tor_mac=active_tor_mac,
+                                            standby_tor_mac=standby_tor_mac,
+                                            active_tor_ip=active_tor_ip,
+                                            standby_tor_ip=standby_tor_ip,
+                                            inner_dscp=dscp,
+                                            outer_dscp=0,
+                                            dst_ip=dst_port_ip
+                                        )
+                pg_shared_wm_res_base = sai_thrift_read_pg_shared_watermark(self.client, asic_type, port_list[src_port_slice])
```

Comment:
better to read the pg counters directly using sai_thrift_read_pg_counters instead of looking at wm

Reply:
Actually I tried to read pg counter with `sai_thrift_read_pg_counters`. But the counter didn't work after remapping. For example, we have remapped DSCP 3 to PG 2, but the counter for PG 2 didn't increase. We still got increment on the original PG3

Reply:
@bingwang-ms , we should check with brcm on this. the correct pg counters should increase

Reply:
> @bingwang-ms , we should check with brcm on this. the correct pg counters should increase

Sure, I have opened a ticket for this issue.